service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ApplicantController.java

```diff
@@ -50,6 +62,16 @@ public AboutToStartOrSubmitCallbackResponse handleMidEvent(@RequestBody Callback
         var updatedApplicants = pbaNumberService.update(caseData.getApplicants());
         data.put(APPLICANTS_PROPERTY, updatedApplicants);
 
+        List<String> applicantEmails = getApplicantEmails(caseData.getApplicants());
+        List<String> errors = validateEmailService.validate(applicantEmails, "Applicant");
```

Comment:
looks that these 2 does not work well together. Lets assume that there are 3 applicants:
applicant1: 
   name: smith
   email: xxx test@test.com
 applicant 2:
   name: White
 applicant 3:
   name: Black
   email: incorrent
   
getApplicantEmails method has filter
 .filter(StringUtils::isNotEmpty)  
  which removes applicant 2 as there is no email
  
  then 2 emails are passed to validator and get messages like
  Applicant 1: .....
  Applicant 2: .....
  
  but Applicant 2 is ok, should be Applicant 3

Reply:
good catch, have changed to work with this

Reply:
fixed this